### PR TITLE
parquet: Add tests for page pruning on unsigned integers

### LINKED ISSUE #9887: parquet: Page pruning on unsigned integers is not tested
### Describe the bug

https://github.com/apache/arrow-datafusion/pull/9778 added tests to datafusion/core/tests/parquet/mod.rs for signed integers , but not for unsigned integers

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by progval**
## Which issue does this PR close?

Closes #9887.

## What changes are included in this PR?

New test scenario, new tests adapted from the signed integer tests

## Are these changes tested?

yes

## Are there any user-facing changes?

no

### CODE REVIEW

File: datafusion/core/tests/parquet/page_pruning.rs [hunk 1]
```diff
@@ -515,6 +515,120 @@ int_tests!(16);
...
+            async fn [<prune_uint $bits _scalar_fun >]() {
+                test_prune(
+                    Scenario::UInt,
+                    &format!("SELECT * FROM t where power(u{}, 2) = 25", $bits),
+                    Some(0),
+                    Some(0),
+                    2,
+                )
+                .await;
+            }
```
Comment (progval) [inline datafusion/core/tests/parquet/page_pruning.rs:584]: `abs()` doesn't really make sense on integers, so I replaced it with `power()` when adapting from signed integer tests

File: datafusion/core/tests/parquet/page_pruning.rs [hunk 2]
```diff
@@ -515,6 +515,120 @@ int_tests!(16);
 int_tests!(32);
 int_tests!(64);
 
+macro_rules! uint_tests {
+    ($bits:expr) => {
+        paste::item! {
+            #[tokio::test]
+            //                      null count  min                                       max
+            // page-0                         0  0                                         4
```
Comment (alamb) [inline datafusion/core/tests/parquet/page_pruning.rs:523]: I was confused at first, but the `make_uint_batches(0, 5)` returns `0..5` which has a max of 4. ✅

File: datafusion/core/tests/parquet/page_pruning.rs (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/parquet/page_pruning.rs:584]: makes sense to me

**Review by alamb — APPROVED** [review]
👨‍🍳 👌  -- thank you @progval . Very nice sequence of tests

**Pull request merged**